### PR TITLE
Use interpolated strings 11

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Replace `string.Format()` with new `string.Create((IFormatProvider? provider, [InterpolatedStringHandlerArgument(nameof(provider))] ref DefaultInterpolatedStringHandler handler)`.

## PR Context
#18974

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/System.Management.Automation/engine/parser/DebugViewWriter.cs [hunk 1]
```diff
@@ -447,12 +447,7 @@ private enum Flow {
 
         protected override Expression VisitLambda<T>(Expression<T> node) {
             Out(
-                string.Format(CultureInfo.CurrentCulture,
-                    "{0} {1}<{2}>",
-                    ".Lambda",
-                    GetLambdaName(node),
-                    node.Type.ToString()
-                )
+                string.Create(CultureInfo.CurrentCulture, $"{".Lambda"} {GetLambdaName(node)}<{node.Type.ToString()}>")
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/parser/DebugViewWriter.cs:450]:
```suggestion
                string.Create(CultureInfo.CurrentCulture, $"{".Lambda"} {GetLambdaName(node)}<{node.Type}>")
```

File: src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs [hunk 2]
```diff
@@ -75,14 +75,7 @@ static OutOfProcessUtils()
...
         {
-            string result = string.Format(CultureInfo.InvariantCulture,
-                "<{0} {1}='{2}' {3}='{4}'>{5}</{0}>",
-                PS_OUT_OF_PROC_DATA_TAG,
-                PS_OUT_OF_PROC_STREAM_ATTRIBUTE,
-                streamType.ToString(),
-                PS_OUT_OF_PROC_PSGUID_ATTRIBUTE,
-                psGuid.ToString(),
-                Convert.ToBase64String(data));
+            string result = string.Create(CultureInfo.InvariantCulture, $"<{PS_OUT_OF_PROC_DATA_TAG} {PS_OUT_OF_PROC_STREAM_ATTRIBUTE}='{streamType.ToString()}' {PS_OUT_OF_PROC_PSGUID_ATTRIBUTE}='{psGuid.ToString()}'>{Convert.ToBase64String(data)}</{PS_OUT_OF_PROC_DATA_TAG}>");
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs:78]:
```suggestion
            string result = string.Create(CultureInfo.InvariantCulture, $"<{PS_OUT_OF_PROC_DATA_TAG} {PS_OUT_OF_PROC_STREAM_ATTRIBUTE}='{streamType}' {PS_OUT_OF_PROC_PSGUID_ATTRIBUTE}='{psGuid}'>{Convert.ToBase64String(data)}</{PS_OUT_OF_PROC_DATA_TAG}>");
```

File: src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs [hunk 3]
```diff
@@ -131,11 +124,7 @@ internal static string CreateSignalAckPacket(Guid psGuid)
         /// <returns></returns>
         private static string CreatePSGuidPacket(string element, Guid psGuid)
         {
-            string result = string.Format(CultureInfo.InvariantCulture,
-                "<{0} {1}='{2}' />",
-                element,
-                PS_OUT_OF_PROC_PSGUID_ATTRIBUTE,
-                psGuid.ToString());
+            string result = string.Format(CultureInfo.InvariantCulture, $"<{element} {PS_OUT_OF_PROC_PSGUID_ATTRIBUTE}='{psGuid.ToString()}' />");
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs:127]:
```suggestion
            string result = string.Format(CultureInfo.InvariantCulture, $"<{element} {PS_OUT_OF_PROC_PSGUID_ATTRIBUTE}='{psGuid}' />");
```

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/parser/DebugViewWriter.cs [hunk 4]
```diff
@@ -447,12 +447,7 @@ private enum Flow {
 
         protected override Expression VisitLambda<T>(Expression<T> node) {
             Out(
-                string.Format(CultureInfo.CurrentCulture,
-                    "{0} {1}<{2}>",
-                    ".Lambda",
-                    GetLambdaName(node),
-                    node.Type.ToString()
-                )
+                string.Create(CultureInfo.CurrentCulture, $"{".Lambda"} {GetLambdaName(node)}<{node.Type}>")
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/parser/DebugViewWriter.cs:450]:
```suggestion
                string.Create(CultureInfo.CurrentCulture, $".Lambda {GetLambdaName(node)}<{node.Type}>")
```

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by iSazonov** [comment]
@CarloToso  You seem to have updated everything possible?

**Comment by CarloToso** [comment]
@iSazonov I think so, this was my regex search query in vscode: `string\.Format.*\{0\}|string\.Create.*\{0\}|\.Append.*\{0\}|\.AppendFormat.*\{0\}|string\.Format.*\n.*\{0\}|string\.Create.*\n.*\{0\}|\.Append.*\n.*\{0\}|\.AppendFormat.*\n.*\{0\}`

**Comment by iSazonov** [comment]
@CarloToso Many thanks!

**Pull request merged**